### PR TITLE
Add missing __LONG_MAX definition in alltypes.h

### PR DESCRIPTION
**Pull request by sbc100**
This mirrors an upstream change in musl
(https://github.com/emscripten-core/musl/commit/7cc79d10) and should really have been done as part of #13006.

With had report from a user who was injecting emscripten's `sysroot/include` path explicitly putting it before the clang's builtin include path which triggered `__LONG_MAX` to be undefined.

### CODE REVIEW

File: system/lib/libc/musl/arch/emscripten/bits/limits.h [hunk 1]
```diff
@@ -1,12 +1 @@
...
 #define PAGE_SIZE 65536
-#define LONG_BIT 32
-#endif
-
-#if __LP64__
-#define LONG_MAX  0x7fffffffffffffffL
-#else
-#define LONG_MAX  0x7fffffffL
-#endif
-#define LLONG_MAX  0x7fffffffffffffffLL
```
Comment (kripken) [inline system/lib/libc/musl/arch/emscripten/bits/limits.h:12]: We don't need to define this in the other header? I tried to find where it was defined elsewhere and saw this:
```
$ grep -U2 -r LLONG_MAX system/lib/libc/musl/include/
system/lib/libc/musl/include/limits.h-#define LONG_MAX __LONG_MAX
system/lib/libc/musl/include/limits.h-#define ULONG_MAX (2UL*LONG_MAX+1)
system/lib/libc/musl/include/limits.h:#define LLONG_MIN (-LLONG_MAX-1)
system/lib/libc/musl/include/limits.h:#define LLONG_MAX  0x7fffffffffffffffLL
system/lib/libc/musl/include/limits.h:#define ULLONG_MAX (2ULL*LLONG_MAX+1)
system/lib/libc/musl/include/limits.h-
system/lib/libc/musl/include/limits.h-#define MB_LEN_MAX 4
```
Does that work? It uses `LLONG_MAX` before it defines it...

Comment (kripken) [reply system/lib/libc/musl/arch/emscripten/bits/limits.h:12]: otherwise lgtm

Comment (sbc100) [reply system/lib/libc/musl/arch/emscripten/bits/limits.h:12]: Seems to work fine yes.   At least this is the same as upstream musl.

Comment (sbc100) [reply system/lib/libc/musl/arch/emscripten/bits/limits.h:12]: Seems like this is just how the C pre-processor works, macros get a expanded at the point of use:

```
#include <stdio.h>

// BAR is not defined at the point of defining FOO
#define FOO BAR

int main() {
#define BAR 12
  printf("FOO=%d\n", FOO); // prints 12
#define BAR 13
  printf("FOO=%d\n", FOO); // prints 13
}
```

Comment (kripken) [reply system/lib/libc/musl/arch/emscripten/bits/limits.h:12]: Oh, really... very interesting. I did not know that about C macros!

**Review by kripken — APPROVED** [review]

File: system/lib/libc/musl/arch/emscripten/bits/alltypes.h [hunk 2]
```diff
@@ -4,6 +4,14 @@
 
 #define __BYTE_ORDER __LITTLE_ENDIAN
 
+// Can't use __LONG_MAX__ here since musl's libs does pre-processor comparison
+// with 0x7fffffffL directly.
```
Comment (arichardson) [inline system/lib/libc/musl/arch/emscripten/bits/alltypes.h:8]: comparing in the pre-processor should work fine even with `#define LONG_MAX __LONG_MAX__`? https://godbolt.org/z/zonffPn6o

Comment (sbc100) [reply system/lib/libc/musl/arch/emscripten/bits/alltypes.h:8]: Hmm, in that case there must be some other reason.. I just assumed that was it.

Comment (sbc100) [reply system/lib/libc/musl/arch/emscripten/bits/alltypes.h:8]: https://github.com/emscripten-core/emscripten/pull/20756

**Pull request merged**